### ISSUE TITLE
Indexing Field Keys

### ISSUE THREAD

**Issue by TechMachineX** [ngx-formly/ngx-formly#1857]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ x ] support request
```

**Current behavior**
We have been using Formly 5.0.0 for a while now, and have come to point where we can start thinking about upgrading to the latest version.

Unfortunately we have found that our existing model structure no longer works in v5.4.3

In v5.0.0 we use the square bracket notation for indexing field keys and correctly creates arrays for our fields, seen here: https://stackblitz.com/edit/angular-wz7kit

in v5.4.3 (and as early as 5.0.1 as far as we can see) this is no longer the case as it doesn't create arrays for the fields, but appends the square brackets to the key, example here: https://stackblitz.com/edit/angular-vvwk8v

I have found that using `.<index>.` e.g. `myParent.0.myfield` syntax works as intended.

**Expected behavior**
Allow square brackets notation for indexing keys.

**Minimal reproduction of the problem with instructions**
working (5.0.0) - https://stackblitz.com/edit/angular-wz7kit
broken (5.4.3) - https://stackblitz.com/edit/angular-vvwk8v

alternative solution (would like to avoid this if possible due to existing code - https://stackblitz.com/edit/angular-y7mskw

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **Angular version:** 2.0.X
Angular 8.2

**Comment by aitboudad** [comment]
[v5.5.2](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.5.2) is released which includes a fix for this issue!

**Comment by TechMachineX** [comment]
looks like its not working when used as the key for repeat components: https://stackblitz.com/edit/angular-vvwk8v

**Comment by aitboudad** [comment]
This second issue has been fixed and released as part of [v5.5.4](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.5.4).

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
